iotilecore/iotile/core/hw/hwmanager.py

```diff
@@ -785,7 +786,52 @@ def get_proxy(self, short_name):
         if short_name not in self._name_map:
             return None
 
-        return self._name_map[short_name][0]
+        proxy_match = self.find_correct_proxy_version(self._name_map[short_name], version)
+        return proxy_match if proxy_match is not None else self._name_map[short_name][0] 
+
+    def find_correct_proxy_version(self, proxies, version):
+        """Retrieves the ModuleVersion of each proxy and match it with the tile version
+
+        something
+
+        Args:
+            proxies (list): A list of proxies of a specific short name
+            version (obj): A tuple that specifies the tile's version
+        """
+
+        proxy_details = {}
+        tile_version = SemanticVersion(version[0], version[1], version[2])
+        min_version = SemanticVersion(0, 0, 0)
+        best_proxy = None
+        self.logger.debug("Short name matched proxies found: {0}".format(proxies))
+        for proxy in proxies:
+            proxy_details[proxy] = {}
+            try:
+                # If proxy has ModuleVersion(), get the SemanticVersionRange
+                module_version = proxy.ModuleVersion()
+                least_version = module_version._disjuncts[0][0][0]
+            except AttributeError:
+                # If proxy does not have ModuleVersion(), use None
+                module_version = None
+                least_version = SemanticVersion(0, 0, 0)
+
+            if module_version is None:
+                # Regardless if version is compatible, choose a best proxy for now
+                if min_version == SemanticVersion(0, 0, 0):
+                    best_proxy = proxy
+                    self.logger.debug("Using default proxy: {0}".format(proxy))
```

Comment:
I might rephrase this to say "Found a proxy without ModuleVersion info". The main culprit word here is "using" -- since we aren't necessarily using it, just found one

Reply:
Gotcha. Fixed.